quinn-proto/src/connection/mod.rs

```diff
@@ -233,6 +233,74 @@ pub struct Connection {
     stats: ConnectionStats,
     /// QUIC version used for the connection.
     version: u32,
+
+    /// qlog streamer to ouput events as JSON text sequences
+    #[cfg(feature = "qlog")]
+    qlog_streamer: Option<qlog::streamer::QlogStreamer>,
+
+    /// Snapshot of the qlog recovery metrics
+    #[cfg(feature = "qlog")]
+    recovery_metrics: CongestionMetrics,
+}
+
+#[cfg(feature = "qlog")]
+#[derive(Default, PartialEq)]
+pub struct CongestionMetrics {
+    pub congestion_window: Option<u64>,
+    pub min_rtt: Option<Duration>,
+    pub smoothed_rtt: Option<Duration>,
+    pub latest_rtt: Option<Duration>,
+    pub rtt_variance: Option<Duration>,
+    pub pto_count: Option<u32>,
+    pub bytes_in_flight: Option<u64>,
+    pub packets_in_flight: Option<u64>,
+    pub ssthresh: Option<u64>,
+    pub pacing_rate: Option<u64>,
```

Comment:
A lot of these metrics don't seem to be the business of the congestion controller to track or report. Should we split out a separate struct that is exclusively congestion-controller-related for the benefit of the trait method? That would help keep the public interface narrower.